src/ussd_public.js

```diff
@@ -0,0 +1,501 @@
+                    new Choice('', $("Trusted friend"))
+                ],
+                next: 'state_msg_receiver_confirm'
+            });
+        });
+
+        // EndState st-11
+        self.add('state_msg_receiver_confirm', function(name) {
+            return new EndState(name, {
+                text: $(questions[name])
+            });
+        });
+
+        // ChoiceState st-12
+        self.add('state_msg_language', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('english', $("English")),
+                    new Choice('hausa', $("Hausa")),
+                    new Choice('igbo', $("Igbo"))
+                ],
+                next: 'state_msg_language_confirm'
+            });
+        });
+
+        // EndState st-13
+        self.add('state_msg_language_confirm', function(name) {
+            return new EndState(name, {
+                text: $(questions[name])
+            });
+        });
+
+        // ChoiceState st-14
+        self.add('state_optout_reason', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('state_msg_loss_subscription', $("Mother miscarried")),
+                    new Choice('state_msg_loss_subscription', $("Baby stillborn")),
+                    new Choice('state_msg_loss_subscription', $("Baby passed away")),
+                    new Choice('state_msg_end_subscription', $("Messages not useful")),
+                    new Choice('state_msg_end_subscription', $("Other"))
+                ],
+                next: function(choice) {
+                    return choice.value;
+                }
+            });
+        });
+
+        // ChoiceState st-15
+        self.add('state_msg_loss_subscription', function(name) {
+            return new ChoiceState(name, {
+                question: $(questions[name]),
+                error: $(get_error_text(name)),
+                choices: [
+                    new Choice('yes', $("Yes")),
+                    new Choice('no', $("No"))
+                ],
+                next: function(choice) {
+                    if(choice.value === 'yes') {
+                        return 'state_msg_loss_subscription_confirm';
+                    }
+                    else {
+                        return 'state_msg_end_subscription';
+                    }
+                }
+            });
+        });
+
+        // EndState st-16
+        self.add('state_msg_loss_subscription_confirm', function(name) {
+            return new EndState(name, {
+                text: $(questions[name])
+            });
+        });
+
+        // EndState st-17
+        self.add('state_msg_end_subscription', function(name) {
```

Comment:
Perhaps call this `state_end_optout`. Generally I like to name Endstates `state_end_...` for easy reference.